flake.nix

```diff
@@ -0,0 +1,144 @@
+{
+  description = "systemd-status-leds - Monitor systemd service status on WS281x LED strips";
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";
+    flake-utils.url = "github:numtide/flake-utils";
+    rust-overlay.url = "github:oxalica/rust-overlay";
+    mcp-nixos.url = "github:utensils/mcp-nixos";
+  };
+
+  outputs = { self, nixpkgs, flake-utils, rust-overlay, mcp-nixos }:
+    flake-utils.lib.eachDefaultSystem (system:
+      let
+        overlays = [ (import rust-overlay) ];
+        pkgs = import nixpkgs {
+          inherit system overlays;
+        };
+
+        # Rust toolchain
+        rustToolchain = pkgs.rust-bin.stable.latest.default.override {
+          extensions = [ "rust-src" "rustfmt" "clippy" ];
+        };
+
+        # Build inputs for the application
+        buildInputs = with pkgs; [
+          # System dependencies
+          pkg-config
+          systemd
+          dbus
+          
+          # Development tools
+          rustToolchain
+          cargo-watch
+          cargo-edit
+          cargo-auditable  # For embedding audit info in binaries
+          
+          # Testing tools
+          cargo-tarpaulin  # For coverage
```

Comment:
What is the current test coverage as a percentage and why are the parts of code currently not tested missing coverage?

Also ain't we missing NixOS modules / tests to deploy the service and test it's working?